samples/custom_dispatch/cpu/embedded/CMakeLists.txt

```
@@ -4,31 +4,38 @@
 # See https://llvm.org/LICENSE.txt for license information.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-if(NOT IREE_TARGET_BACKEND_LLVM_CPU OR
-   NOT IREE_HAL_DRIVER_LOCAL_SYNC OR
+if(NOT IREE_HAL_DRIVER_LOCAL_SYNC OR
    NOT IREE_HAL_EXECUTABLE_LOADER_EMBEDDED_ELF)
   return()
 endif()
 
-# This only builds for x86-64 because this is just a sample and we don't feature
-# detect what backends are compiled into clang. We could extend this to build
+# This only builds for x86-64 and arm64. We could extend this to build
 # for the current cmake target architecture but would also need to modify the
 # MLIR file to have the new target configuration.
-if(NOT IREE_ARCH STREQUAL "x86_64")
-  message(STATUS "IREE custom_dispatch/cpu/embedded ignored -- only builds for x86_64 (today)")
+iree_compiler_targeting_iree_arch(_BUILD_COMPILER_TARGETING_X86_64 "x86_64")
+iree_compiler_targeting_iree_arch(_BUILD_COMPILER_TARGETING_ARM_64 "arm_64")
+# First check compiler support
+if(NOT (_BUILD_COMPILER_TARGETING_X86_64 AND _BUILD_COMPILER_TARGETING_ARM_64))
+  message(STATUS "IREE custom_dispatch/cpu/embedded ignored - some required LLVM targets are not built.")
+  return()
+endif()
+# Check if the host architecture matches what we are building code for.
+if(NOT (IREE_ARCH STREQUAL "x86_64" OR IREE_ARCH STREQUAL "arm_64"))
+  message(STATUS "IREE custom_dispatch/cpu/embedded ignored - host architecture unsupported.")
```

nit: similar condition change? (up to you)

```suggestion
if(NOT IREE_ARCH STREQUAL "x86_64" AND NOT IREE_ARCH STREQUAL "arm_64")
  message(STATUS "IREE custom_dispatch/cpu/embedded ignored - host architecture unsupported.")
```